tests/shared/test_exporter.py

```diff
@@ -0,0 +1,57 @@
+import os
+import pytest
+import shutil
+from ert_shared.exporter import Exporter
+from tests.utils import SOURCE_DIR, tmpdir
+from res.enkf import ResConfig, EnKFMain
+from ert_shared.cli.notifier import ErtCliNotifier
+from ert_shared import ERT
+from ert_shared.plugins.plugin_manager import ErtPluginContext
+
+
+@tmpdir(os.path.join(SOURCE_DIR, "test-data/local/snake_oil"))
+def test_exporter_is_valid():
+    with ErtPluginContext():
+        config_file = "snake_oil.ert"
+        rc = ResConfig(user_config_file=config_file)
+        rc.convertToCReference(None)
+        ert = EnKFMain(rc)
+        notifier = ErtCliNotifier(ert, config_file)
+        ERT.adapt(notifier)
+        ex = Exporter()
+        assert ex.is_valid()
+
+
+@tmpdir(os.path.join(SOURCE_DIR, "test-data/local/snake_oil"))
+def test_exporter_is_not_valid():
+    config_file = "snake_oil.ert"
+    rc = ResConfig(user_config_file=config_file)
+    rc.convertToCReference(None)
+    ert = EnKFMain(rc)
+    notifier = ErtCliNotifier(ert, config_file)
+    ERT.adapt(notifier)
+    ex = Exporter()
+    assert not ex.is_valid()
+
+
+@tmpdir(os.path.join(SOURCE_DIR, "test-data/local/snake_oil"))
+def test_run_export():
+    with ErtPluginContext():
+        config_file = "snake_oil.ert"
+        rc = ResConfig(user_config_file=config_file)
+        rc.convertToCReference(None)
+        ert = EnKFMain(rc)
+        notifier = ErtCliNotifier(ert, config_file)
+        ERT.adapt(notifier)
+        ex = Exporter()
+        parameters = {
+            "output_file": "export.csv",
+            "time_index": "raw",
+            "column_keys": "FOPR",
+        }
+        ex.run_export(parameters)
```

Comment:
When this fails due to missing semeio, it will fail like this:

`AttributeError: 'NoneType' object has no attribute 'run'` at `export_job.run(…)`.

Could perhaps assert that `ex.is_valid()` with a message?

Reply:
I dont see why that would not happen?